### DISCUSSION TITLE
Connect AD users with already existing user in Kimai Database

### DISCUSSION THREAD

**Discussion by WDI-Ideas** [kimai/kimai#4855] [Questions & Answers]
Dear kevinpapst,

We were using kimai in installed in a VPS server for some months and it has user data in it. Users were created in Kimai Local database (mysql)
 
 Currently i am shifting it to local server and trying to connect it with AD users.
 I successfully connected new user created in AD and logged in as a new user in kimai.
 But I have to create AD users and connect the user with already existing user without losing the data of the user.
 I tried creating the same username and password in AD and changed the auth from kimai to ldap in user database. but no success.
 
 Can we achieve this in kimai . if yes kindly share what other changes to be done to achieve this.
 
 Kimai version : 2.0.16

**Comment by kevinpapst** [comment]
Nothing has to be changed, the user needs to have the same identifiers in both systems, namely email and username, depends on your setup.

**Reply by WDI-Ideas** [reply]
I have kept Email, Username and Password same, but log shows : 

**Failed Already existing User :**

 app.ERROR: ldap_bind(RajendraReddy, ****) {"action":"ldap_bind","bindDn":"RajendraReddy"} []
 security.INFO: Authenticator failed. {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\BadCredentialsException(code: 0): Fetching user data/roles failed, probably DN is expired. at /var/www/kimai/src/Ldap/LdapCredentialsSubscriber.php:76)","authenticator":"App\\Ldap\\LdapAuthenticator"} []

**PASSED TEST USER:**

 app.ERROR: ldap_bind(testuser, ****) {"action":"ldap_bind","bindDn":"testuser"} []

 app.ERROR: ldap_search(dc=internal,dc=wdipl,dc=com, (&(&(objectClass=user))(sAMAccountName=testuser)), array["+","*"]) {"action":"ldap_search","base_dn":"dc=internal,dc=wdipl,dc=com","filter":"(&(&(objectClass=user))(sAMAccountName=testuser))","attributes":["+","*"]} []

app.ERROR: ldap_search(CN=test user,OU=Server-Team,DC=internal,DC=wdipl,DC=com, (objectClass=user), array["+","*"]) {"action":"ldap_search","base_dn":"CN=test user,OU=Server-Team,DC=internal,DC=wdipl,DC=com","filter":"(objectClass=user)","attributes":["+","*"]} []

security.INFO: Authenticator successful! {"token":{"Symfony\\Component\\Security\\Core\\Authentication\\Token\\UsernamePasswordToken":"UsernamePasswordToken(user=\"testuser\", roles=\"ROLE_USER\")"},"authenticator":"App\\Ldap\\LdapAuthenticator"} []

Can you suggest any step for troubleshooting this as even after keeping the credentials same , it not logging in
